app/routes.py

```diff
@@ -1 +1,237 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+import requests
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+def validate_model_by_id(cls, model_id):
+    '''Helper function to validate database IDs prior to executing queries.'''
+    try:
+        model_id = int(model_id)
+    except ValueError:
+        return abort(make_response({"msg": f"{model_id} is not valid"}, 400))
+    
+    object = cls.query.get(model_id)
+
+    if object is None:
+        return abort(make_response({"msg": f"{model_id} not found"}, 404))
+
+    return object
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    '''Gets all tasks from the task table in the database.'''
+    sort_query = request.args.get("sort")
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+    
+    response = []
+    for task in tasks:
+            response.append(task.to_dict())
+    
+    return jsonify(response), 200
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_one_task():
+    '''Creates a new task, requires title and description, id will be autogenerated.'''
+    request_body = request.get_json()
+
+    try:
+        new_task = Task(title=request_body["title"], description=request_body["description"])
+            
+    except Exception as e:
+        return jsonify({"details": "Invalid data"}), 400
+
```

Comment:
Nice use of a try-except, I would suggest some way to give the user feedback about what was wrong in the data.  Maybe tell them if the title or description was wrong.